Polyfills/DOM/Source/DOM.cpp

```diff
@@ -0,0 +1,69 @@
+#include <Babylon/Polyfills/DOM.h>
+
+namespace Babylon::Polyfills::DOM
+{
+    class DOMPointReadOnly final : public Napi::ObjectWrap<DOMPointReadOnly>
+    {
+        static constexpr auto JS_CLASS_NAME = "DOMPointReadOnly";
+
+    public:
+        static void Initialize(Napi::Env env) {
+            Napi::HandleScope scope{ env };
+
+            Napi::Function func = DefineClass(
+                env,
+                JS_CLASS_NAME,
+                {
+                    InstanceAccessor("x", &DOMPointReadOnly::GetX, nullptr),
+                    InstanceAccessor("y", &DOMPointReadOnly::GetY, nullptr),
+                    InstanceAccessor("z", &DOMPointReadOnly::GetZ, nullptr),
+                    InstanceAccessor("w", &DOMPointReadOnly::GetW, nullptr)
+                });
+
+            env.Global().Set(JS_CLASS_NAME, func);
+        }
+
+        DOMPointReadOnly(const Napi::CallbackInfo& info) : Napi::ObjectWrap<DOMPointReadOnly>{info}
+        {
+            _x = info[0].As<Napi::Number>();
+            _y = info[1].As<Napi::Number>();
```

Comment:
according to spec, even the first two values are optional, so you probably need to check the length for >0 and >1 also.